source/extensions/common/workload_discovery/discovery.proto

```diff
@@ -27,64 +27,152 @@ option go_package = "test/envoye2e/workloadapi";
+// they can be used for Gateways.
+// In this case, the key format will be "network/hostname".
+// These resources cannot be looked up on-demand.
+message Address {
+  oneof type {
+    // Workload represents an individual workload.
+    // This could be a single Pod, a VM instance, etc.
+    Workload workload = 1;
+    // Service represents a service - a group of workloads that can be accessed together.
+    Service service = 2;
+  }
+}
+
+// Service represents a service - a group of workloads that can be accessed together.
+// The primary key is "namespace/hostname".
+// Secondary (alias) keys are the unique `network/IP` pairs that the service can be reached at.
+message Service {
+  // Name represents the name for the service.
+  // For Kubernetes, this is the Service name.
+  string name = 1;
+  // Namespace represents the namespace for the service.
+  string namespace = 2;
+  // Hostname represents the FQDN of the service.
+  // For Kubernetes, this would be <name>.<namespace>.svc.<cluster domain>
+  // TODO: support this field
+  string hostname = 3;
+  // Address represents the addresses the service can be reached at.
+  // There may be multiple addresses for a single service if it resides in multiple networks,
+  // multiple clusters, and/or if it's dual stack (TODO: support dual stack).
+  reserved "addresses";
+  reserved 4;
+  // Ports for the service.
+  // The target_port may be overridden on a per-workload basis.
+  reserved "ports";
+  reserved 5;
+  // Optional; if set, the SAN to verify for TLS connections.
+  // Typically, this is not set and per-workload identity is used to verfiy
+  // TODO: support this field
+  repeated string subject_alt_names = 6;
+}
+
 message Workload {
+  // UID represents a globally unique opaque identifier for this workload.
+  string uid = 20;
   // Name represents the name for the workload.
   // For Kubernetes, this is the pod name.
   // This is just for debugging and may be elided as an optimization.
   string name = 1;
-
   // Namespace represents the namespace for the workload.
   // This is just for debugging and may be elided as an optimization.
   string namespace = 2;
 
-  // IP address of the workload.
+  // Address represents the IPv4/IPv6 address for the workload.
+  // This should be globally unique.
+  // This should not have a port number.
+  // Each workload must have at least either an address or hostname; not both.
+  // TODO: support dual stack by making this repeated
+  // TODO: when this is repeated, update xds primary key from network/IP to network/UID
   bytes address = 3;
 
+  // The hostname for the workload to be resolved by the ztunnel.
+  // DNS queries are sent on-demand by default.
+  // If the resolved DNS query has several endpoints, the request will be forwarded
+  // to the first response.
+  // Each workload must have at least either an address or hostname; not both.
+  // TODO: support this field
+  string hostname = 21;
```

Comment:
All these fields are not needed and just cause overhead. Can you just reserve their fields to avoid warnings and not define them?